### ISSUE TITLE
Unable to delete user accounts

### ISSUE THREAD

**Issue by Gamebuster19901** [Gamebuster19901/WilderForge#32]
When attempting to delete a profile, the following exception occurs
```
[11:47:24.783] [main/ERROR] [FaultTolerantInputMultiplexer/]: Unhandled exception in response to user input! java.lang.NullPointerException
	at com.worldwalkergames.legacy.context.ClientDataContext.updateAccountAlwaysOnModsLists(ClientDataContext.java:1288)
	at com.worldwalkergames.legacy.control.AccountClient.selectAccount(AccountClient.java:43)
	at com.worldwalkergames.legacy.control.ClientControl.deletePlayer(ClientControl.java:293)
	at com.worldwalkergames.legacy.ui.menu.PlayerSelect$2.onAffirmative(PlayerSelect.java:162)
	at com.worldwalkergames.legacy.ui.MessageBoxDialog.affirmative(MessageBoxDialog.java:146)
	at com.worldwalkergames.communication.observer.signals.Signal2.dispatch(Signal2.java:170)
	at com.worldwalkergames.legacy.game.common.ui.OptionButton.tryClick(OptionButton.java:116)
	at com.worldwalkergames.legacy.game.common.ui.OptionButton$1.clicked(OptionButton.java:97)
	at com.badlogic.gdx.scenes.scene2d.utils.ClickListener.touchUp(ClickListener.java:89)
	at com.badlogic.gdx.scenes.scene2d.InputListener.handle(InputListener.java:69)
	at com.badlogic.gdx.scenes.scene2d.Stage.touchUp(Stage.java:352)
	at com.badlogic.gdx.InputMultiplexer.touchUp(InputMultiplexer.java:124)
	at com.worldwalkergames.legacy.ui.FaultTolerantInputMultiplexer.touchUp(FaultTolerantInputMultiplexer.java:63)
	at com.badlogic.gdx.InputEventQueue.drain(InputEventQueue.java:86)
	at com.badlogic.gdx.backends.lwjgl3.DefaultLwjgl3Input.update(DefaultLwjgl3Input.java:208)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Window.update(Lwjgl3Window.java:394)
	at com.badlogic.gdx.backends.lwjgl3.NiceLwjgl3Application.loop(NiceLwjgl3Application.java:104)
	at com.badlogic.gdx.backends.lwjgl3.NiceLwjgl3Application.<init>(NiceLwjgl3Application.java:80)
	at com.worldwalkergames.legacy.LegacyDesktop.main(LegacyDesktop.java:117)
	at com.wildermods.wilderforge.launch.Main.launchGame(Main.java:70)
	at com.wildermods.wilderforge.launch.Main.main(Main.java:38)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.wildermods.wilderforge.launch.services.WilderLauncherService.lambda$0(WilderLauncherService.java:68)
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37)
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54)
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72)
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:82)
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:66) 
```

**Comment by Gamebuster19901** [comment]
This is an issue in the vanilla game.

**Comment by Gamebuster19901** [comment]
Vanilla stacktrace (java 16)

```
0:37:548 4 FaultTolerantInputMultiplexer Unhandled exception in response to user input! java.lang.NullPointerException: Cannot read field "activeMods" because "account" is null
	at com.worldwalkergames.legacy.context.ClientDataContext.updateAccountAlwaysOnModsLists(ClientDataContext.java:1288)
	at com.worldwalkergames.legacy.control.AccountClient.selectAccount(AccountClient.java:43)
	at com.worldwalkergames.legacy.control.ClientControl.deletePlayer(ClientControl.java:293)
	at com.worldwalkergames.legacy.ui.menu.PlayerSelect$2.onAffirmative(PlayerSelect.java:162)
	at com.worldwalkergames.legacy.ui.MessageBoxDialog.affirmative(MessageBoxDialog.java:146)
	at com.worldwalkergames.communication.observer.signals.Signal2.dispatch(Signal2.java:170)
	at com.worldwalkergames.legacy.game.common.ui.OptionButton.tryClick(OptionButton.java:116)
	at com.worldwalkergames.legacy.game.common.ui.OptionButton$1.clicked(OptionButton.java:97)
	at com.badlogic.gdx.scenes.scene2d.utils.ClickListener.touchUp(ClickListener.java:89)
	at com.badlogic.gdx.scenes.scene2d.InputListener.handle(InputListener.java:69)
	at com.badlogic.gdx.scenes.scene2d.Stage.touchUp(Stage.java:352)
	at com.badlogic.gdx.InputMultiplexer.touchUp(InputMultiplexer.java:124)
	at com.worldwalkergames.legacy.ui.FaultTolerantInputMultiplexer.touchUp(FaultTolerantInputMultiplexer.java:63)
	at com.badlogic.gdx.InputEventQueue.drain(InputEventQueue.java:86)
	at com.badlogic.gdx.backends.lwjgl3.DefaultLwjgl3Input.update(DefaultLwjgl3Input.java:208)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Window.update(Lwjgl3Window.java:394)
	at com.badlogic.gdx.backends.lwjgl3.NiceLwjgl3Application.loop(NiceLwjgl3Application.java:104)
	at com.badlogic.gdx.backends.lwjgl3.NiceLwjgl3Application.<init>(NiceLwjgl3Application.java:80)
	at com.worldwalkergames.legacy.LegacyDesktop.main(LegacyDesktop.java:117)

```

**Comment by Gamebuster19901** [comment]
Reported on wildermyth discord

**Comment by Gamebuster19901** [comment]
This issue has been resolved upstream - From 1.4 patch notes:

```
Fix a bug where deleting an account with mods would crash
```
